isis/src/socet/apps/socetframesettings/socetframesettings.cpp

```diff
@@ -471,110 +683,284 @@ void getCamPosOPK(Spice &spice, QString spacecraftName, double et, Camera *cam,
+  //////////////////////////////////////////////////////////////////////
+  // End of section setting isisFocalPlane2SocetPlate
+  //////////////////////////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////
+
+  //____________________________________________________________________
+  // From Camera object, fetch rotation matrices that convert vectors from
+  // J2000 inertial frame to Ocentric frame and to ISIS Focal Plane frame.
+  // - (Planet-)Ocentric => target body-fixed [+X = PMxEq.; +Z = TPRP]
+  // - PMxEq. => intersection of target Prime Merdian and Equator
+  // - TPRP => Target Positive Rotation Pole, typically = North
+  vector<double>  j2000ToOcentricMatrixVector = cam->bodyRotation()->Matrix();
+  vector<double>  j2000ToIsisFocalPlaneMatrixVector = cam->instrumentRotation()->Matrix();
+
+  // Reformat rotation matrices from 9-element vector<double>'s to 3x3 arrays
+  SpiceDouble j2000ToOcentricRotationMatrix[3][3] = MTX3x3_ZEROS;
+  SpiceDouble j2000ToIsisFocalPlaneMatrix[3][3] = MTX3x3_ZEROS;
+
   for (int j = 0; j < 3; j++) {
     for (int k = 0; k < 3; k++) {
-      j2000ToBodyFixedRotationMatrix[j][k] = j2000ToBodyFixedMatrixVector[3 * j + k];
-      j2000ToCameraRotationMatrix[j][k] = j2000ToCameraMatrixVector[3 * j + k];
+      j2000ToOcentricRotationMatrix[j][k] = j2000ToOcentricMatrixVector[3 * j + k];
+      j2000ToIsisFocalPlaneMatrix[j][k] = j2000ToIsisFocalPlaneMatrixVector[3 * j + k];
     }
   }
 
-  // Compute Camera to Body Fixed rotation matrix
-  double cameraToBodyFixedRotationMatrix[3][3] = {{0.0, 0.0, 0.0},
-                                                  {0.0, 0.0, 0.0},
-                                                  {0.0, 0.0, 0.0}};
-  mxmt_c(j2000ToBodyFixedRotationMatrix, j2000ToCameraRotationMatrix,
-         cameraToBodyFixedRotationMatrix);
+  // Compute rotation matrix from ISIS Focal Plane frame to Ocentric frame
+  SpiceDouble isisFocalPlaneToOcentricRotationMatrix[3][3] = MTX3x3_ZEROS;
+  mxmt_c(j2000ToOcentricRotationMatrix, j2000ToIsisFocalPlaneMatrix,
+         isisFocalPlaneToOcentricRotationMatrix);
 
-  // get instrumemt position vector and planet radii in meters
-  double instrumentPosition[3] = {0.0, 0.0, 0.0};
+  // Get instrumemt position vector, convert to meters
+  SpiceDouble instrumentPosition[3] = VEC3_ZEROS;
   spice.instrumentPosition(instrumentPosition);
-  for (int i = 0; i < 3; i++) {
-    instrumentPosition[i] *= 1000.0;
-  }
+  vscl_c(1000.0, instrumentPosition, instrumentPosition);
 
+  // Get planet radii
   Distance dRadii[3];
   spice.radii(dRadii);
-  double radii[3] = {0.0, 0.0, 0.0};
-  radii[0] = dRadii[0].meters();
-  radii[1] = dRadii[1].meters();
-  radii[2] = dRadii[2].meters();
 
-  // Calculate ographic coordinates of spacecraft position vector
+  // Calculate ographic coordinates of spacecraft position vector, in meters
+  SpiceDouble equatorRadius_m = dRadii[0].meters();
```

Comment:
ISIS standards do not use underscore in local variable names. equatorialRadius_m should be changed to something like equatorialRadiusMeters.

Reply:
agreed.